base/util/PKICertImport.bash

```diff
@@ -258,38 +495,316 @@ function PKICertImport() {
     # Remove the certificate from the NSS DB specified by $1. Errors are fatal;
     # uses exit instead of return.
     function _remove_cert() {
-        local database="$1"
+        local nickname="$1"
+        local return_no_exit="$2"
+
         local remove_args=("-D")
 
-        remove_args+=("-d" "$NSSDB_TYPE$database")
+        remove_args+=("-d" "$NSSDB_TYPE$NSSDB")
         if [ "x$NSSDB_PASSWORD" != "x" ]; then
             remove_args+=("-f" "$NSSDB_PASSWORD")
         fi
 
-        __v certutil "${remove_args[@]}" "-n" "$CERT_NICKNAME"
-        certutil "${remove_args[@]}" "-n" "$CERT_NICKNAME"
+        __v certutil "${remove_args[@]}" "-n" "$nickname"
+        certutil "${remove_args[@]}" "-n" "$nickname"
         local ret=$?
         if (( ret != 0 )); then
             __e "certutil returned non-zero result: $ret"
             __e "Unable to delete certificate!"
-            exit $?
+
+            if [ "x$return_no_exit" != "xtrue" ]; then
```

Comment:
I'm not sure if we also need to detect HSMs here...

I'm wondering in particular about when we have a HSM and delete a certificate on the token before deleting a token on the HSM... Wondering if we want to try both and if either fails, return an error? Thoughts?